presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -74,7 +74,7 @@
 
     Optional<TableHandle> getTableHandleForStatisticsCollection(Session session, QualifiedObjectName tableName, Map<String, Object> analyzeProperties);
 
-    List<TableLayoutResult> getLayouts(Session session, TableHandle tableHandle, Constraint<ColumnHandle> constraint, Optional<Set<ColumnHandle>> desiredColumns);
+    Optional<TableLayoutResult> getLayout(Session session, TableHandle tableHandle, Constraint<ColumnHandle> constraint, Optional<Set<ColumnHandle>> desiredColumns);
```

Comment:
It should return `TableLayoutResult`, without the `Optional`.

- Connectors never supposed to return an empty list
- This check should be directly moved to the `PickTableLayout`:
   ```
        if (constraint.getSummary().isNone()) {
            return Optional.empty();
        }
    ```